docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+points for the `santa_monica`, `CA` location than for the others. The differing number of values 
+will be instructive later when we shape and summarize this data.
+
+#### Summarizing and Grouping with Aggregates
+Let's get into an example for summarizing some of our water data. For the sake of these queries, assume that 
+we're executing them at `2020-02-22T18:00`, that way we can use relative times in the query. Say we want to get the 
+last 3 hours of data, with the min, max, and mean of `bottom_degrees` for that time period. With that 
+definition, we'd expect to get 1 row per series in the dataset (assuming that we have min, max, and mean as 
+columns in the resulting table. Here's the query:
+
+```
+@my_data.h2o_temperature{bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}) 
+```
+
+Will return the following dataset:
+
+| min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees |
+| --- | --- | --- |
+| 48 | 51 | 47.45 |
+
+Even though the sample data set has three series, we get back only a single row that gives the summary statistics we 
+requested. The call to the aggregate function passed in an **Aggregate Summary Block** which looks similar in 
+construction to the **Predicate Filter Block** that we used on the first line of the query. The returned column names 
+are given automatically based on the construction `<function name>_<field name>`. As with the other blocks, field 
+names can be wrapped in double quotes to handle cases like spaces or other special characters in a field name.
+
+Our results don't have the descriptive tag metadata because we didn't ask for it. Selecting `location` and `state` 
+which are the two tags that we have for data in the `h20_measurements` measurement will ensure it comes through in 
+the aggregate and we can then group by those columns:
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["location", "state"]) 
+```
+
+| location | state | min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees |
+| --- | --- | --- | --- | --- |
+| coyote_creek | CA | 48 | 50 | 48.714 |
+| puget_sound | CA | 40 | 40 | 40 |
+| santa_monica | WA | 50 | 51 | 50.5 |
+
+Although we could have left out the `state` grouping from our example because in this case the `location` is 
+enough to unique identify a series.
+
+Say we wanted to get a summary per state: 
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["state"]) 
+```
+
+| state | min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees |
+| --- | --- | --- | --- |
+| CA | 48 | 51 | 49.11 |
+| WA | 40 | 40 | 40 |
+
+See that the results don't have the `location` column even though we selected it on the first line. That's because 
```

Comment:
This is really confusing IMO. To select a column and not have it returned. I think I'd rather the query error tbh.

Reply:
Would this query give the same result?
```
@my_data.h2o_temperature{time > -3h}
    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["state"]) 
```

Reply:
This would be an error in SQL .. something like "not in aggregate function or group by clause"